src/services/GISService/index.ts

```diff
@@ -0,0 +1,246 @@
+import Service from '../Service'
+import {GISResult, IrismonumentProperties, MunicipalityStatistics, Statistics} from './types/Irismonument'
+import { Params } from './types/Params'
+
+const URL_BASE = 'https://gis.urban.brussels/geoserver/ows'
+
+class GISSerice extends Service {
+
+  static ParamsBuilder = class {
+    private params: Params
+
+    constructor() {
+      this.params = {
+        service: 'wfs',
+        version: '2.0.0',
+        request: 'GetFeature',
+        TypeName: 'BSO_DML_BESC:Inventaris_Irismonument', // 'BDU_DMS_PROT:Inventaire_Irismonument',
+        outputformat: 'application/json',
+        cql_filter: {},
+        srsname: 'EPSG:4326'
+      }
+    }
+
+    setService (service: string): this {
+      this.params.service = service
+      return this
+    }
+
+    setVersion (version: string): this {
+      this.params.version = version
+      return this
+    }
+
+    setRequest (request: string): this {
+      this.params.request = request
+      return this
+    }
+
+    setTypeName (typeName: string): this {
+      this.params.TypeName = typeName
+      return this
+    }
+
+    setOutputFormat (outFormat: string): this {
+      this.params.outputformat = outFormat
+      return this
+    }
+
+    setCQLFilter (cqlFilter: IrismonumentProperties): this {
+      this.params.cql_filter = cqlFilter
+      return this
+    }
+
+    setSRSName (name: string): this {
+      this.params.srsname = name
+      return this
+    }
+
+    build (strict = true): string {
```

Comment:
I suggest you use `URLSearchParams` to build the string.
See https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams